odoo/modules/loading.py

```diff
@@ -88,15 +88,15 @@ def _get_files_of_kind(kind: str) -> list[str]:
     return bool(filename)
 
 
-def load_demo(env: Environment, package: Node, idref, mode: str) -> bool:
+def load_demo(env: Environment, package: Package, idref, mode: str) -> bool:
     """
     Loads demo data for the specified package.
     """
-    if not package.should_have_demo():
+    if tools.config['without_demo'] or package.state != 'to install' or not package.demo_installable():
```

Comment:
while it still can be discussed about the demo installation

But for this problem, since `demo_installable()` will dynamically check if its parents' demo data are installed I think it is not a problem. And depended module will be checked after the its parent module when checking `demo_installable()`, its parents demo should already been installed (package.demo will be changed after `load_demo`)